MobileBuy/sample/src/main/java/com/shopify/sample/activity/AndroidPayCheckoutActivity.java

```diff
@@ -0,0 +1,271 @@
+
+    private MaskedWallet maskedWallet;
+
+    private GoogleApiClient googleApiClient;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        maskedWallet = getSampleApplication().getMaskedWallet();
+
+        setTitle(R.string.checkout);
+        setContentView(R.layout.android_pay_checkout_activity);
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+
+        Button confirmationButton = (Button) findViewById(R.id.confirm_button);
+        confirmationButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                showLoadingDialog(R.string.completing_checkout);
+                Checkout checkout = getSampleApplication().getCheckout();
+                MaskedWallet maskedWallet = getSampleApplication().getMaskedWallet();
+
+                FullWalletRequest fullWalletRequest = AndroidPayHelper.createFullWalletRequest(checkout, maskedWallet);
+                Wallet.Payments.loadFullWallet(googleApiClient, fullWalletRequest, AndroidPayHelper.REQUEST_CODE_FULL_WALLET);
+            }
+        });
+
+        updateOrderSummary();
+
+        createGoogleAPIClient();
+
+        createAndAddWalletFragment();
+    }
+
+    private void createAndAddWalletFragment() {
+
+        WalletFragmentStyle walletFragmentStyle = new WalletFragmentStyle()
+               .setMaskedWalletDetailsHeaderTextAppearance(R.style.SampleWalletFragmentDetailsHeaderTextAppearance)
+               .setMaskedWalletDetailsTextAppearance(R.style.SampleWalletFragmentDetailsTextAppearance)
+               .setMaskedWalletDetailsBackgroundColor(android.R.color.white);
+
+        WalletFragmentOptions walletFragmentOptions = WalletFragmentOptions.newBuilder()
+                .setEnvironment(SampleApplication.WALLET_ENVIRONMENT)
+                .setFragmentStyle(walletFragmentStyle)
+                .setTheme(WalletConstants.THEME_LIGHT)
+                .setMode(WalletFragmentMode.SELECTION_DETAILS)
+                .build();
+
+        SupportWalletFragment walletFragment = SupportWalletFragment.newInstance(walletFragmentOptions);
+
+        WalletFragmentInitParams.Builder startParamsBuilder = WalletFragmentInitParams.newBuilder()
+                .setMaskedWallet(maskedWallet)
+                .setMaskedWalletRequestCode(AndroidPayHelper.REQUEST_CODE_CHANGE_MASKED_WALLET);
+        walletFragment.initialize(startParamsBuilder.build());
+
+        // add Wallet fragment to the UI
+        getSupportFragmentManager().beginTransaction()
+                .replace(R.id.google_wallet_fragment_container, walletFragment)
+                .commit();
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+
+        int errorCode = data.getIntExtra(WalletConstants.EXTRA_ERROR_CODE, 0);
+
+        switch (requestCode) {
```

Comment:
I would suggest to extract this logic to helper, and provide call back with meaningful names something like these:
onMaskedWalletChange
onCheckoutConfirm
onCheckoutCancel